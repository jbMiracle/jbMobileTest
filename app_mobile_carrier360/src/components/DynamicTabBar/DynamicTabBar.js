import React from 'react';
import { TabBarTop, TabNavigator, StackNavigator } from 'react-navigation';
import C360Colors from '../../themes/360Colors';
import styles from './DynamicTabBar.style';
import LoadDetails from '../../screens/MyLoads/LoadDetails/LoadDetails';
import OfferDetails from '../../screens/Offers/OfferDetails/OfferDetails';

export default (tabs, parent) => {
  const initialRouteName = tabs[0].screenName;
  const sharedHeaderConfig = { headerMode: 'none' };

  const tabRoutes = {};
  tabs.forEach((screen) => {
    tabRoutes[screen.screenName] = { screen: screen.component };
  });

  const DynamicTabs = TabNavigator(
    tabRoutes,
    {
      initialRouteName,
      tabBarPosition: 'top',
      swipeEnabled: false,
      animationEnabled: true,
      tabBarComponent: props => (
        <TabBarTop { ...props } indicatorStyle={ styles.indicatorStyle } />
      ),
      tabBarOptions: {
        style: styles.tabBarStyle,
        tabStyle: styles.tabStyle,
        activeTintColor: C360Colors.White,
        upperCaseLabel: false,
        labelStyle: styles.labelStyle,
      },
    }
  );
  const finalStack = { DynamicTabs };

  switch (parent) {
    case 'MyLoads':
      finalStack['Load Details'] = { screen: LoadDetails };
      break;
    case 'Offers':
      finalStack['Offer Details'] = { screen: OfferDetails };
      break;
    default:
      break;
  }

  return StackNavigator(finalStack, sharedHeaderConfig);
};
