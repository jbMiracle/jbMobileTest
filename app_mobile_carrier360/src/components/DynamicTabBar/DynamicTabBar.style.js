import { StyleSheet } from 'react-native';
import C360Colors from '../../themes/360Colors';

export default StyleSheet.create({
  tabBarStyle: {
    borderTopWidth: 0,
    backgroundColor: C360Colors.Blue500,
  },
  tabStyle: {
    paddingHorizontal: 0,
  },
  labelStyle: {
    fontSize: 12,
    fontWeight: 'bold',
  },
  indicatorStyle: {
    backgroundColor: '#FAD802',
    height: 4,
  },
});
