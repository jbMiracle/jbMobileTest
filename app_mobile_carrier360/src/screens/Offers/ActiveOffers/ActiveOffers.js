import React, { Component } from 'react';
import { View } from 'react-native';
import { Text, Button } from 'native-base';
import C360Colors from '../../../themes/360Colors';

class ActiveOffers extends Component {
  render() {
    return (
      <View testID='offers' style={ { flex: 1, justifyContent: 'center', alignItems: 'center' } }>
        <Text>Active Offers</Text>
        <Button
          style={ {
              backgroundColor: C360Colors.Yellow500, padding: 0, marginTop: 15, alignSelf: 'center',
          } }
          onPress={ () => this.props.navigation.navigate('Offer Details', { msg: 'Here are the details...' }) }
        >
          <Text style={ { color: C360Colors.Blue600, fontSize: 12, fontWeight: 'bold' } }>Show Offer Details</Text>
        </Button>
      </View>
    );
  }
}

export default ActiveOffers;
