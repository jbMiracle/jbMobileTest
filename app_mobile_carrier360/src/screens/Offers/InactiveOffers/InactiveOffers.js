import React, { Component } from 'react';
import { Text, View } from 'react-native';

class InactiveOffers extends Component {
  render() {
    return (
      <View testID='offers' style={ { flex: 1, justifyContent: 'center', alignItems: 'center' } }>
        <Text>Inactive Offers</Text>
      </View>
    );
  }
}

export default InactiveOffers;
