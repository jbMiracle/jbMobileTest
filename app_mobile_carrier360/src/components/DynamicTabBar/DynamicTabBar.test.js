import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import DynamicTabBar from '../../components/DynamicTabBar/DynamicTabBar';
import ActiveOffers from '../../screens/Offers/ActiveOffers/ActiveOffers';
import InactiveOffers from '../../screens/Offers/InactiveOffers/InactiveOffers';

it('can render header tabs', () => {
  const renderer = new ShallowRenderer();
  const tabs = [
    { screenName: 'Active Offers', component: ActiveOffers },
    { screenName: 'Inactive Offers', component: InactiveOffers },
  ];
  DynamicTabBar(tabs);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
