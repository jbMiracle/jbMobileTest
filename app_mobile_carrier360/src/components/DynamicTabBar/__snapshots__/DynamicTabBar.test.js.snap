// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render header tabs 1`] = `null`;
